sdk/samples/iot/hub/src/sample_pnp_thermostat_component.c

```diff
@@ -0,0 +1,363 @@
+static const az_span report_max_temp_name_span = AZ_SPAN_LITERAL_FROM_STR("maxTemp");
+static const az_span report_min_temp_name_span = AZ_SPAN_LITERAL_FROM_STR("minTemp");
+static const az_span report_avg_temp_name_span = AZ_SPAN_LITERAL_FROM_STR("avgTemp");
+static const az_span report_start_time_name_span = AZ_SPAN_LITERAL_FROM_STR("startTime");
+static const az_span report_end_time_name_span = AZ_SPAN_LITERAL_FROM_STR("endTime");
+static const az_span report_error_payload = AZ_SPAN_LITERAL_FROM_STR("{}");
+static char end_time_buffer[32];
+static char incoming_since_value[32];
+
+// Twin Values
+static const az_span desired_temp_property_name = AZ_SPAN_LITERAL_FROM_STR("targetTemperature");
+static const az_span temp_response_description_success = AZ_SPAN_LITERAL_FROM_STR("success");
+static const az_span temp_response_description_failed = AZ_SPAN_LITERAL_FROM_STR("failed");
+static const az_span max_temp_reported_property_name
+    = AZ_SPAN_LITERAL_FROM_STR("maxTempSinceLastReboot");
+
+// ISO8601 Time Format
+static const char iso_spec_time_format[] = "%Y-%m-%dT%H:%M:%S%z";
+
+static az_result build_command_response_payload(
+    sample_pnp_thermostat_component* handle,
+    az_json_writer* json_builder,
+    az_span start_time_span,
+    az_span end_time_span,
+    az_span* response_payload)
+{
+  double avg_temp = handle->device_temperature_avg_total / handle->device_temperature_avg_count;
+  // Build the command response payload
+  AZ_RETURN_IF_FAILED(az_json_writer_append_begin_object(json_builder));
+  AZ_RETURN_IF_FAILED(az_json_writer_append_property_name(json_builder, report_max_temp_name_span));
+  AZ_RETURN_IF_FAILED(az_json_writer_append_double(
+      json_builder, handle->max_temperature, DOUBLE_DECIMAL_PLACE_DIGITS));
+  AZ_RETURN_IF_FAILED(az_json_writer_append_property_name(json_builder, report_min_temp_name_span));
+  AZ_RETURN_IF_FAILED(az_json_writer_append_double(
+      json_builder, handle->min_temperature, DOUBLE_DECIMAL_PLACE_DIGITS));
+  AZ_RETURN_IF_FAILED(az_json_writer_append_property_name(json_builder, report_avg_temp_name_span));
+  AZ_RETURN_IF_FAILED(
+      az_json_writer_append_double(json_builder, avg_temp, DOUBLE_DECIMAL_PLACE_DIGITS));
+  AZ_RETURN_IF_FAILED(
+      az_json_writer_append_property_name(json_builder, report_start_time_name_span));
+  AZ_RETURN_IF_FAILED(az_json_writer_append_string(json_builder, start_time_span));
+  AZ_RETURN_IF_FAILED(az_json_writer_append_property_name(json_builder, report_end_time_name_span));
+  AZ_RETURN_IF_FAILED(az_json_writer_append_string(json_builder, end_time_span));
+  AZ_RETURN_IF_FAILED(az_json_writer_append_end_object(json_builder));
+
+  *response_payload = az_json_writer_get_bytes_used_in_destination(json_builder);
+
+  return AZ_OK;
+}
+
+// Invoke the command requested from the service. Here, it generates a report for max, min, and avg
+// temperatures.
+static az_result invoke_getMaxMinReport(
```

Comment:
```suggestion
static az_result invoke_getMaxMinReport_command(
```

Reply:
?

Reply:
ehh other samples have it as this so I'm partial to leave it for now